Tests/UnitTests/ess-download-exchangeset.component.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { EssDownloadExchangesetComponent } from '../../src/app/features/exchange-set/ess-download-exchangeset/ess-download-exchangeset.component';
+import { AppConfigService } from '../../src/app/core/services/app-config.service';
+import { EssUploadFileService } from '../../src/app/core/services/ess-upload-file.service';
+import { CommonModule } from '@angular/common';
+import { Router } from '@angular/router';
+import { of } from 'rxjs';
+
+describe('EssDownloadExchangesetComponent', () => {
+  let component: EssDownloadExchangesetComponent;
+  let fixture: ComponentFixture<EssDownloadExchangesetComponent>;
+  const router = {
+    navigate: jest.fn()
+  };
+
+  const service = {
+    getValidEncs : jest.fn().mockReturnValue(['AU210130', 'AU210140', 'AU220130', 'AU220150', 'AU314128']),
+    clearSelectedEncs : jest.fn(),
+    getSelectedENCs: jest.fn(),
+    infoMessage : true,
+    addSelectedEnc : jest.fn(),
+    removeSelectedEncs : jest.fn(),
+    getNotifySingleEnc : jest.fn().mockReturnValue(of(true)),
+    getExchangeSetDetails: jest.fn().mockReturnValue({exchangeSetCellCount : 4}),
+    exchangeSetCreationResponse: jest.fn().mockReturnValue(of(exchangeSetDetailsMockData)),
+    getEstimatedTotalSize:jest.fn()
+  };
+  
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [CommonModule],
+      declarations: [ EssDownloadExchangesetComponent ],
+      providers: [
+        {
+          provide : EssUploadFileService,
+          useValue : service
+        },
+        {
+          provide: Router,
+          useValue: router
+        }
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    AppConfigService.settings = {
+      essConfig: {
+      MaxEncLimit: 100,
+      MaxEncSelectionLimit : 5,
+      avgSizeofENC: 0.3
```

Comment:
Please change to `avgSizeofENCinMB`
```suggestion
      avgSizeofENCinMB: 0.3
```